src/server/zset_family.cc

```diff
@@ -691,23 +690,39 @@ void InterScoredMap(ScoredMap* dest, ScoredMap* src, AggType agg_type) {
     dest->swap(*src);
 }
 
-OpResult<ScoredMap> OpUnion(EngineShard* shard, Transaction* t, string_view dest, AggType agg_type,
-                            const vector<double>& weights, bool store) {
-  ArgSlice keys = t->ShardArgsInShard(shard->shard_id());
-  DVLOG(1) << "shard:" << shard->shard_id() << ", keys " << vector(keys.begin(), keys.end());
-  DCHECK(!keys.empty());
-
-  unsigned start = 0;
+template <typename SliceIt>
+ScoredMap UnionSliceResultScore(SliceIt start, SliceIt end, AggType agg_type) {
+  ScoredMap result;
+  for (auto it = start; it != end; ++it) {
+    if (it->first.is_done()) {
+      continue;
+    }
 
-  if (keys.front() == dest) {
-    ++start;
+    ScoredMap sm = FromObject(it->first->second, it->second);
+    if (result.empty()) {
+      result.swap(sm);
+    } else {
+      UnionScoredMap(&result, &sm, agg_type);
+    }
   }
+  return result;
+}
+
+using ScoredPI = pair<PrimeIterator, double>;
```

Comment:
The only place using this types are in the FindSliceResult func definition. I think it will be more readable to define
auto FindSliceResult(..) -> vector<pair<PrimeIterator, double>>

Reply:
Actually you could define this type and then use it in serveral places:
1. The function definition
2. For the arr_it var inside the function
3. For the definition of UnionSliceResultScore paras instead the template typename

I would call it UnionElementsWeightVec because it used for holding the items and their corresponding weights in a vector